src/addressBook/index.js

```diff
@@ -337,13 +369,13 @@ export function createContact(addressBookIri, contact, types) {
       return (t) => addUrl(t, vcard.hasTelephone, [iri, name].join("#"));
     })
   );
-
-  const dataset = [...emails, ...addresses, ...telephones].reduce(
+  let dataset = [...emails, ...addresses, ...telephones].reduce(
     (acc, { thing }) => {
       return setThing(acc, thing);
     },
     person
   );
+  dataset = webIdNode && setThing(dataset, webIdNode);
```

Comment:
I'm a little concerned about this line, as it makes the overall logic a bit hard to follow for me. I reckon if `webIdNode` is falsy, it will simply return `false` for dataset? I cannot see any test that back up this scenario, or am I misunderstanding this?

Reply:
I can't remember exactly why I added that check tbh. Removed it since the webIdNode is never undefined (there's always a webId when we add a contact, and if it's invalid it will fail to fetch the profile and won't add the contact anyway).